### ISSUE TITLE
Discussion: .env/.env.example and activist dev onboarding

### ISSUE THREAD

**Issue by andrewtavis** [activist-org/activist#570]
### Terms

- [X] I have searched [open and closed issues](https://github.com/activist-org/activist/issues?q=is%3Aissue)
- [X] I agree to follow activist's [Code of Conduct](https://github.com/activist-org/activist/blob/main/.github/CODE_OF_CONDUCT.md)

### Issue

We discussed in Code Night 🌙 that the `.env` file inheriting from `.env.example` is problematic as if we make changes to the base file a contributor won't necessarily know to update their environment. Let's discuss how this could be improved!

**Comment by andrewtavis** [comment]
CC @to-sta and @wkyoshida :)

**Comment by andrewtavis** [comment]
Assigning the three of us so we don't forget about. Noticed it again as I was helping someone onboard today :) Let's maybe get this done in the next FlokiNET call?

**Comment by to-sta** [comment]
We could add `env_file:` to read the `.env.example` file in the `docker-compose.yaml` file.

**Comment by andrewtavis** [comment]
Do we do that or just change `.env` to be that? Will have a `.env` file in production that would have different values? Do we need that if we have multiple instances of the Docker up and running at a given time?

**Comment by to-sta** [comment]
> Do we do that or just change `.env` to be that? 

I would suggest to keep the .env ignored amd follow this pattern:

```
.env
.env.*
!.env.example
```

> Will have a `.env` file in production that would have different values?

We can use the `.env` file in production and  if it is necessary `.env.staging` for staging.

> Do we need that if we have multiple instances of the Docker up and running at a given time?

Hmm.. so when we have different docker-compose files for dev and prod, we can also load different env. files for each of them. 

For prod we could use .env and create the file on the server. 

Hope that makes sense.

**Comment by andrewtavis** [comment]
> Hope that makes sense.

Yes very much so, @to-sta :) Thanks for the confirmation! I'll make the changes tomorrow 😴

P.S. I think you'd enjoy Congress. Not sure if you'd have interest, but would be cool to have you up here for the next one. There's been lots of talk about how we should do a general table where people can ask questions, a workshop or even lightning talk if it gets accepted at the next one 😊

**Comment by andrewtavis** [comment]
Hey @to-sta 👋 Realized we didn't check on this and in looking at it now, where should I put ` env_file: .env.example`? I'm getting that it's not allowed in the syntax?

**Comment by to-sta** [comment]
hmm 🤔... How did you implement it? As far as I know we have to declare the env_file for every service

```yaml
// ...

services:
     backend:
         env_file:
             - ./.env.example
      // other services
```

**Comment by andrewtavis** [comment]
Doesn't seem to be working, @to-sta:

```bash
activist-org/activist [main●] » docker-compose up --build
WARN[0000] The "DATABASE_NAME" variable is not set. Defaulting to a blank string. 
WARN[0000] The "DATABASE_USER" variable is not set. Defaulting to a blank string. 
WARN[0000] The "DATABASE_PASSWORD" variable is not set. Defaulting to a blank string. 
WARN[0000] The "DATABASE_PORT" variable is not set. Defaulting to a blank string. 
WARN[0000] The "DATABASE_PORT" variable is not set. Defaulting to a blank string. 
WARN[0000] The "BACKEND_PORT" variable is not set. Defaulting to a blank string. 
WARN[0000] The "BACKEND_PORT" variable is not set. Defaulting to a blank string. 
WARN[0000] The "BACKEND_PORT" variable is not set. Defaulting to a blank string. 
WARN[0000] The "DATABASE_NAME" variable is not set. Defaulting to a blank string. 
WARN[0000] The "DATABASE_USER" variable is not set. Defaulting to a blank string. 
WARN[0000] The "DATABASE_PASSWORD" variable is not set. Defaulting to a blank string. 
WARN[0000] The "DATABASE_HOST" variable is not set. Defaulting to a blank string. 
WARN[0000] The "DATABASE_PORT" variable is not set. Defaulting to a blank string. 
WARN[0000] The "DJANGO_ALLOWED_HOSTS" variable is not set. Defaulting to a blank string. 
WARN[0000] The "DEBUG" variable is not set. Defaulting to a blank string. 
WARN[0000] The "SECRET_KEY" variable is not set. Defaulting to a blank string. 
WARN[0000] The "FRONTEND_PORT" variable is not set. Defaulting to a blank string. 
WARN[0000] The "FRONTEND_PORT" variable is not set. Defaulting to a blank string. 
WARN[0000] The "FRONTEND_PORT" variable is not set. Defaulting to a blank string. 
```

The above is post making the change in `backend`, `frontend` and `db` of the `docker-compose.yml` and then deleting `.env` 🤔

**Comment by andrewtavis** [comment]
Thanks for continuing to look into this :)

**Comment by andrewtavis** [comment]
Closed via afc5fb9. @to-sta, @wkyoshida and I took a look at this and what needed to change was it has to be:

```
env_file:
   - .env.example
```

Not `- ./.env.example` 😊 We set all of this up with a `.env.dev` that we thought was a bit more explicit, and removed the directions to copy it over to `.env`. Thanks both for the work and discussion here!
